submarine-workbench/workbench-server/src/main/java/org/apache/submarine/database/utils/HttpRequestUtil.java

```diff
@@ -0,0 +1,187 @@
+    httpURLConnection.setRequestProperty("accept", "application/json");
+    httpURLConnection.setRequestProperty("connection", "Keep-Alive");
+    httpURLConnection.setRequestProperty("user-agent",
+        "Mozilla/4.0 (compatible; MSIE 6.0; Windows NT 5.1;SV1)");
+    // Timestamp (accurate to seconds)
+    httpURLConnection.setRequestProperty("timestamp", String.valueOf(new Date().getTime() / 1000));
+
+    if (headParams != null) {
+      for (String key : headParams.keySet()) {
+        httpURLConnection.setRequestProperty(key, headParams.get(key));
+      }
+    }
+    httpURLConnection.setUseCaches(false);
+
+    if (parameter != null) {
+      try (BufferedOutputStream outputStream =
+               new BufferedOutputStream(httpURLConnection.getOutputStream())) {
+        outputStream.write(parameter);
+        outputStream.flush();
+      } catch (IOException e) {
+        LOG.error(e.getMessage(), e);
+      }
+    }
+    try (InputStreamReader inputStreamReader =
+             new InputStreamReader(httpURLConnection.getInputStream(), "utf-8");
+         BufferedReader bufferedReader = new BufferedReader(inputStreamReader)) {
+      String line;
+      while ((line = bufferedReader.readLine()) != null) {
+        stringBuilder.append(line);
+      }
+    } catch (IOException e) {
+      LOG.error(e.getMessage(), e);
+    }
+
+    return stringBuilder.toString();
+  }
+
+  public static void main(String[] args) {
+    String editParameter = "{\n" +
+        "\t\"id\": \"0adf4dfca00111e5a63d00ff4d7a64c3\",\n" +
+        "\t\"name\": \"test_a\",\n" +
+        "\t\"userName\": \"userName\",\n" +
+        "\t\"description\": \"test_a\",\n" +
+        "\t\"type\": \"PROJECT_TYPE_NOTEBOOK\",\n" +
+        "\t\"teamName\": \"\",\n" +
+        "\t\"visibility\": \"PROJECT_VISIBILITY_PRIVATE\",\n" +
+        "\t\"permission\": \"PROJECT_PERMISSION_VIEW\",\n" +
+        "\t\"starNum\": 0,\n" +
+        "\t\"likeNum\": 0,\n" +
+        "\t\"messageNum\": 0,\n" +
+        "\t\"createBy\": \"userName\",\n" +
+        "\t\"updateBy\": \"userName\"\n" +
+        "}";
+    String addParameter = "{\n" +
+        "\"name\": \"test_project2\",\n" +
+        "\"visibility\": \"PROJECT_VISIBILITY_PRIVATE\",\n" +
+        "\"type\": \"PROJECT_TYPE_NOTEBOOK\",\n" +
+        "\"permission\": \"PROJECT_PERMISSION_VIEW\",\n" +
+        "\"tags\": null,\n" +
+        "\"starNum\": 0,\n" +
+        "\"likeNum\": 0,\n" +
+        "\"messageNum\": 0,\n" +
+        "\"teamName\": \"\",\n" +
+        "\"description\": \"test_project2\",\n" +
+        "\"userName\": \"test\",\n" +
+        "\"projectFilesList\": [],\n" +
+        "\"createBy\": \"test\",\n" +
+        "\"createTime\": \"2019-10-16 15:58:52\",\n" +
+        "\"updateBy\": \"test\",\n" +
+        "\"updateTime\": \"2019-10-16 15:58:52\"\n" +
+        "}";
+
+    String deleteParameter = "3ff46684a03011e5a63d00ff4d7a64c3";
+    try {
+      // String result =
```

Comment:
@LinhaoZhu  there are a lot of comment code, how about refactor the code to test case?

Reply:
Because running this part of the functionality requires github token and the corresponding test repository, no runnable test cases are submitted.

Reply:
We can create test ropo in `hadoopsubmarine`. This code needs to move to test case.

Reply:
Done.